driver/README.API_VERSION

```diff
@@ -0,0 +1,11 @@
+# API version number
+
+The file API_VERSION must contain a semver-compatible version number of the userspace<->kernel API. All other lines are ignored.
+
+## When to increment
+
+**major version**: increment when the probe becomes incompatible with previous userspace versions
+
+**minor version**: increment when new features are added but existing features remain compatible (e.g. new event fields or new events)
+
+**patch version**: increment when code changes don't break compatibility (e.g. bug fixes, new kernel version support)
```

Comment:
@leogr @FedeDP I think we need to change this. If probe 1.0.5 is not compatible with userspace 1.0.6 then we definitely should not bump the api version for (build time) new kernel support, otherwise it defeats the whole point.

We'd only need to bump the patch level if the probe compiled for a particular kernel but it didn't work correctly.

Thoughts?

Reply:
Ideally, we need two different variables for the **driver API version** and the **scap-supported driver API version**.
Let me explain.

Assuming the driver version is 1.0.5 and the scap supported version is 1.0.5. 
Then we add a new kernel and we bump only the driver version to 1.1.0 and leave the scap supported version to 1.0.5
The version check continues to work as expected.

Later, we discover a bug that need to be fixed both in the driver and scap.
The driver will be bumped to 1.1.1 and scap will start to require version 1.1.1.

I believe this way we get both the expected behavior and respect the version checking semantic.
Basically, we just need two separate version variables for the producer and the consumer. That's what normally happens when the two components live in separate places.

do you like this solution :smiley_cat: ?

Reply:
I typed a long reply here but in the end I realized I agree almost fully with your proposal, except for one thing:

> Then we add a new kernel and we bump only the driver version to 1.1.0

If this new kernel didn't build before, I'd leave the version number alone (there's no way for users of that kernel to run an older version and the new kernel support shouldn't affect any other kernels; if it does, it's a bug, not a bugfix)

Reply:
Not sure I really understood your point. Driver version 1.1.0 is backward compatible with all driver API versions from 1.0.0.
So what's the issue with bumping it when adding new kernel support?
The meaning of bumping from `1.0.5` to `1.1.0` is just to indicate to the consumer that something new was added.

Reply:
Sorry, missed your comment :(

The issue is that then we require API 1.0.(x+1) instead of 1.0.x which requires a rebuild of drivers for all the already supported (and unaffected) kernels, largely defeating the point (scap 1.0.(x+1) will refuse to work with driver 1.0.x).